### ISSUE TITLE
Converted version, e and c and gelan?

### ISSUE THREAD

**Issue by abhaskumarsinha** [WongKinYiu/yolov9#209]
Technically, what is the difference between `yolov9-c-converted`, `yolov9-e-converted`, `yolov9c`, `gelan-c`, `gelan-e`?

From what I understand - Gelan are the inference version of yolov9-c. What is the difference between `c` and `e` editions?
Why loading the weights from yolov9c from yaml have different number of parameters than loading of weights?
What is the meaning of converted version?

**Comment by Youho99** [comment]
Check #131 
gelan is not inference version of yolov9

The GELAN models have the improvement of the GELAN block in the architecture
YOLOv9 models have the improvement of the GELAN block in the architecture, as well as auxiliary branches

From what I understand, the YOLOv9-converted models are YOLOv9 models where the auxiliary branches have been removed, for the inference of these YOLOv9 models (only)

**Comment by abhaskumarsinha** [comment]
Also I forgot to ask, what is this fuse() operation in YOLO.py?

**Comment by Youho99** [comment]
https://stackoverflow.com/questions/56601075/what-is-a-fused-kernel-or-fused-layer-in-deep-learning

**Comment by Youho99** [comment]
Search in the issues, you will find your answer

But from what I understand, yolov9 without aux for training is just gelan

**Comment by abhaskumarsinha** [comment]
Ah okay... Got it!
Just e and c in yolov9 is left.

**Comment by abhaskumarsinha** [comment]
One thing I didn't get is whether one needs to remove aux manually after training or during inference when `self.training = False` for all nn.Modules, it automatically inferences with required modules, dropping aux by itself?

**Comment by WongKinYiu** [comment]
https://github.com/WongKinYiu/yolov9?tab=readme-ov-file#re-parameterization

**Comment by abhaskumarsinha** [comment]
@WongKinYiu So what's the difference between the c and e version of yolov9?

**Comment by Youho99** [comment]
Your answer is in the ReadMe
Number of parameters / architecture

**Comment by abhaskumarsinha** [comment]
Makes sense. I'm closing the ticket, I'm satisfied.

**Comment by WongKinYiu** [comment]
yolov9-c: modified from [yolov7-af](https://openaccess.thecvf.com/content/CVPR2023/html/Wang_YOLOv7_Trainable_Bag-of-Freebies_Sets_New_State-of-the-Art_for_Real-Time_Object_Detectors_CVPR_2023_paper.html)
yolov9-e: modified from [dynamic-yolov7](https://openaccess.thecvf.com/content/CVPR2023/html/Lin_DynamicDet_A_Unified_Dynamic_Architecture_for_Object_Detection_CVPR_2023_paper.html)

**Comment by ZhangJiaJie955** [comment]
> Check #131 gelan is not inference version of yolov9
> 
> The GELAN models have the improvement of the GELAN block in the architecture YOLOv9 models have the improvement of the GELAN block in the architecture, as well as auxiliary branches
> 
> From what I understand, the YOLOv9-converted models are YOLOv9 models where the auxiliary branches have been removed, for the inference of these YOLOv9 models (only)

So, it sounds like there  is not much difference between GELAN and YOLOv9-converted?

**Comment by abhaskumarsinha** [comment]
> > Check #131 gelan is not inference version of yolov9
> > The GELAN models have the improvement of the GELAN block in the architecture YOLOv9 models have the improvement of the GELAN block in the architecture, as well as auxiliary branches
> > From what I understand, the YOLOv9-converted models are YOLOv9 models where the auxiliary branches have been removed, for the inference of these YOLOv9 models (only)
> 
> So, it sounds like there is not much difference between GELAN and YOLOv9-converted?

Here's a summary:

- yolov9 architecture contains GELAN blocks + Aux architecture.
- Aux is only helpful during training and gradient operations and is not useful after once the model is trained.
- yolov9-converted is the architecture where Aux part of the model is removed and simplified.
- GELAN is the version where only GELAN version of the model was training and not whole yolov9 (which includes Aux Architecture).
- Yolov9 has similar architecture to GELAN once Aux is removed, but different weights as when GELAN is trained separately as Aux makes training more efficient.

This is basically what I've gathered till now.

**Comment by MadhavKrishna1** [comment]
@abhaskumarsinha I understood the difference between gelan.pt, yolov9.pt, and yolov9_converted.pt. Now, if I want to fine-tune on a different dataset (say VisDrone), which model should I use? Also, if I want to modify the architecture, which model should I start with?
